src/views/AnnualReport.vue

```diff
@@ -541,177 +589,160 @@ export default {
+          // set the new AGM date in the AGM Date component (may be null or empty)
+          this.newAgmDate = annualReport.annualGeneralMeetingDate || ''
+          // set the new No AGM flag in the AGM Date component (may be undefined)
+          this.newNoAgm = annualReport.didNotHoldAgm || false
         } else {
-          // eslint-disable-next-line no-console
-          console.log('fetchData() error - invalid response =', response)
-          this.resumeErrorDialog = true
+          // otherwise asOfDate will be calculated some other way (see getter)
+        }
+
+        // restore Change of Directors data
+        const changeOfDirectors = filing.changeOfDirectors
+        if (changeOfDirectors) {
+          if (changeOfDirectors.directors?.length > 0) {
+            this.updatedDirectors = changeOfDirectors.directors
+            // NB: filing data will be set by director paid/free events
+          } else {
+            throw new Error('Invalid change of directors object')
+          }
+        } else {
+          // changeOfDirectors is optional
+          // leave existing directors intact
+        }
+
+        // restore Change of Address data
+        const changeOfAddress = filing.changeOfAddress
+        if (changeOfAddress) {
+          // registered office is required
+          // records office is required for BCOMP only
+          const registeredOffice = changeOfAddress.offices?.registeredOffice
+          const recordsOffice = changeOfAddress.offices?.recordsOffice
+          if (this.isBComp && registeredOffice && recordsOffice) {
+            this.updatedAddresses = { registeredOffice, recordsOffice }
+          } else if (!this.isBComp && registeredOffice) {
+            this.updatedAddresses = { registeredOffice }
+          } else {
+            throw new Error('Invalid change of address object')
+          }
+
+          // update filing data
+          // always set Priority flag to false
+          // use existing Waive Fees flag
+          this.updateFilingData('add', FilingCodes.ADDRESS_CHANGE_OT, false,
+            (this.staffPaymentData.option === StaffPaymentOptions.NO_FEE))
+        } else {
+          // changeOfAddress is optional
+          // leave existing office addresses intact
         }
       }).catch(error => {
         // eslint-disable-next-line no-console
-        console.log('fetchData() error =', error)
+        console.log('fetchDraftFiling() error =', error)
         this.resumeErrorDialog = true
       })
     },
 
     /**
      * Callback method for the "modified" event from OfficeAddress.
-     *
      * @param modified a boolean indicating whether or not the office address(es) have been modified from their
      * original values.
      */
     officeModifiedEventHandler (modified: boolean): void {
-      this.haveChanges = true
-      // when addresses change, update filing data
+      // only record changes once the initial data is loaded
+      this.haveChanges = this.dataLoaded
```

Comment:
This is to prevent "you have unsaved data" popup when exiting and having made no change.
In other words, this ignores data changes made by loading/resuming a filing.